src/main/java/frc/robot/poseestimator/PoseEstimationMath.java

```diff
@@ -0,0 +1,166 @@
+package frc.robot.poseestimator;
+
+import edu.wpi.first.math.geometry.*;
+import frc.robot.constants.Field;
+import frc.robot.poseestimator.observations.IRobotPoseVisionObservation;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+public class PoseEstimationMath {
+
+	public static Twist2d addGyroToTwist(Twist2d twist, Rotation2d currentGyroAngle, Rotation2d lastGyroAngle) {
+		boolean hasGyroUpdated = currentGyroAngle != null;
+		if (hasGyroUpdated && lastGyroAngle != null) {
+			return updateChangeInAngle(twist, currentGyroAngle, lastGyroAngle);
+		}
+		return twist;
+	}
```

Comment:
I would suggest maybe to only call this method if the gyro angles are not null, it is not a math thing to check if something is null.
Also, I believe this is copied from 6328 and there is no scenario in which the gyro angle is null. If the gyro is disconnected, the swerve breaks anyway

Reply:
I don't want to trust @YHerm about this. He likes nulls. 
Anyway I changed it to be a little less cursed